sdk/core/core-lro/src/lroEngine/lroEngine.ts

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { Poller } from "../poller";
+import { PollOperationState } from "../pollOperation";
+import { LongRunningOperation, LroEngineOptions, ResumablePollOperationState } from "./models";
+import { GenericPollOperation } from "./operation";
+
+/**
+ * The LRO Engine, a class that performs polling.
+ */
+export class LroEngine<TResult, TState extends PollOperationState<TResult>> extends Poller<
+  TState,
+  TResult
+> {
+  private intervalInMs: number;
+
+  constructor(lro: LongRunningOperation<TResult>, options?: LroEngineOptions) {
+    const { intervalInMs = 2000, resumeFrom } = options || {};
+    function deserializeState(
+      serializedState: string
+    ): TState & ResumablePollOperationState<TResult> {
+      try {
+        return JSON.parse(serializedState).state;
+      } catch (e) {
+        throw new Error(`LroEngine: Unable to deserialize state: ${serializedState}`);
+      }
+    }
+    const state: TState & ResumablePollOperationState<TResult> = resumeFrom
+      ? deserializeState(resumeFrom)
+      : ({} as any);
+
+    const operation = new GenericPollOperation(state, lro, options?.lroResourceLocationConfig);
+    super(operation);
+
+    this.intervalInMs = intervalInMs;
+    operation.setPollerConfig(this as any);
```

Comment:
this looks a little fishy

Reply:
this `intervalInMs` is a private, could this be `operation.setPollerConfig({intervalInMs: this.intervalInMs});` ?

Reply:
I agree it is fishy but I needed to do this because I wanted to be able to set the `intervalInMs` member of `LroEngine` inside the operation using the `retry-after` header value. I am open to ideas to make this cleaner.

Reply:
perhaps you could have a private object, e.g.

```
this.pollerConfig = {
   intervalInMs
}
```

then pass that object along, and since you still have a reference to it, you can update it later.

Reply:
For some reason, I thought `intervalInMs` is a public member so that was why I did it that way. I fixed it now.